File: Data Structures & Algorithms/jump-game/submission-3.java
class Solution {
    public boolean canJump(int[] nums) {
        boolean[] dp = new boolean[nums.length];

        dp[0] = true;
        for(int i = 0; i < nums.length; i++){
            int jumps = nums[i];

            if(dp[i]){
                for(int j = 0; i + j <  dp.length && j <= jumps; j++){
                    
                    dp[i+j] = true;
                }
            }
        }

        return dp[nums.length-1];


    }
}

File: Data Structures & Algorithms/two-integer-sum/submission-3.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        complements = {}
        for i in range(len(nums)):
            complement = target - nums[i]
            if complement in complements:
                return [complements.get(complement), i]
            complements[nums[i]] = i

        return []
        